README.md

```diff
@@ -19,25 +17,78 @@ And then execute:
 Or install it yourself as:
 
     $ gem install event-sourcing
+    
+## Concepts
+
+This frameworks introduces artifacts and rules to keep track of object state changes.
+
+The core artifacts of this framework are:
+
+- **Aggregates**: represent the current state of the application
+    - Usually they are persistence objects, such as "models"
+- **Events**: provide history of an aggregate, applying changes to an aggregate
+    - "Applying" an event is also known as "Calculator" on other event sourcing literature
+- **Commands**: validates and persists events
+- **Reactors**: trigger side effects as events happen
+- **Event Dispatcher**: manage which reactors will be executed after an event happens
+
+While state changing actions obey the rationale:
+
+- Aggregate state changes should be executed through a Command
+- Events are immutable, as they represent series of changes to an Aggregate
+- Reactors may be triggered after an Event occur
 
 ## Usage
 
-TODO: Write usage instructions here
+This framework can be used on any Ruby project, as it allows you to customize how you can handle Event persistence and
+event dispatching control flow.
+
+You can find an implementation sample at the [PORO app's README file](examples/poro_app/README.md).  
+
+## Usage with Rails
+
+You can also use this framework with Rails, as it includes an `event_sourcing/rails` module to ease up configuration.
+
+**Note:** this Rails implementation requires a database that implements serializable attributes by default,
+such as PostgreSQL JSON/Binary JSON columns.
+
+To include the Rails module, add this line to your application's Gemfile:
+
+```ruby
+gem "event-sourcing", require: "event_sourcing/rails"
+```
+
+Further installation steps and a sample implementation can be found at the [Rails' README file](examples/rails_app/README.md).
 
 ## Development
 
-After checking out the repo, run `bin/setup` to install dependencies. Then, run `rake spec` to run the tests. You can also run `bin/console` for an interactive prompt that will allow you to experiment.
+- System requirements
```

Comment:
I would move everything under the "Development" section to "Contributing". Keep things simple.